src/foam/dao/Relationship.js

```diff
@@ -403,21 +403,30 @@ foam.CLASS({
       label: '',
       factory: function() {
         var targetDAO = this.__context__[this.targetDAOKey];
-        foam.assert(targetDAO, 'Missing DAO for targetDAAOKey', this.targetDAOKey);
+        foam.assert(targetDAO, 'Missing DAO for targetDAOKey', this.targetDAOKey);
+
         return foam.dao.ReadOnlyDAO.create({
           delegate: foam.dao.ManyToManyRelationshipDAO.create({
             relationship: this,
             delegate: targetDAO
           }, this)
         }, this);
       },
-      javaFactory: `return new foam.dao.ReadOnlyDAO.Builder(getX()).
-  setDelegate(new foam.dao.ManyToManyRelationshipDAO.Builder(getX()).
-    setRelationship(this).
-    setDelegate((foam.dao.DAO)getX().get(getTargetDAOKey())).
-    build()).
-  build();`,
-
+      javaFactory: `
+        try {
+          return new foam.dao.ReadOnlyDAO.Builder(getX()).
+            setDelegate(new foam.dao.ManyToManyRelationshipDAO.Builder(getX()).
+              setRelationship(this).
+              setDelegate((foam.dao.DAO)getX().get(getTargetDAOKey())).
+              build()).
+            build();
+        } catch ( NullPointerException e ) {
+          foam.nanos.logger.Logger logger = (foam.nanos.logger.Logger) getX().get("logger");
+          logger.error("TargetDAOKey", getTargetDAOKey(), "not found.", e);
```

Comment:
Anyway to output what the relationship is? Else the error is hard to track down.

Reply:
<img width="278" alt="screen shot 2019-02-27 at 3 20 35 pm" src="https://user-images.githubusercontent.com/33228583/53520272-48626900-3aa3-11e9-9512-22bd7da97fed.png">

When I set wrong targetKey and test many to many relationship on User, I got above assert.

Reply:
That's enough, thanks.